docs/service-fabric/refactor-migrated-app.md

```diff
@@ -0,0 +1,155 @@
+
+## Scenario
+
+As discussed in the previous article, [migrating an Azure Cloud Services application to Azure Service Fabric][migrate-from-cloud-services], in 2012 the patterns & practices team documented the process for designing and implementing a Cloud Services application in Azure. The book describes a fictitious company named Tailspin that wants to create a Cloud Services application named **Surveys**. The Surveys application allows users to create and publish surveys that can be answered by the public. The following diagram shows the architecture of this version of the Surveys application:
+
+![](./images/surveys_01.png)
+
+The **Tailspin.Web** web role hosts an ASP.NET MVC site that Tailspin customers use to:
+* sign up for the Surveys application,
+* create or delete a survey,
+* view results for a survey,
+* request that survey results be exported to SQL, and
+* view survey results and analysis.
+
+The **Tailspin.Web.Survey.Public** web role also hosts an ASP.NET MVC site that the public visits to fill out the surveys. Their responses are put in a queue to be saved.
+
+The **Tailspin.Workers.Survey** worker role performs background processing by picking up requests from multiple queues.
+
+Last year, the patterns & practices team created a new project to port this application to Azure Service Fabric. The goal of this project was to make only the necessary code changes to get the application running in an Azure Service Fabric cluster. As a result, the original web and worker roles were not decomposed into a more granular architecture. The resulting architecture is very similar to the Cloud Service version of the application:
+
+![](./images/surveys_02.png)
+
+The **Tailspin.Web** service is ported from the original *Tailspin.Web* web role.
+
+The **Tailspin.Web.Survey.Public** service is ported from the original *Tailspin.Web.Survey.Public* web role.
+
+The **Tailspin.AnswerAnalysisService** service is ported from the original *Tailspin.Workers.Survey* worker role.
+
+> [!NOTE] 
+> While minimal code changes were made to each of the web and worker roles, **Tailspin.Web** and **Tailspin.Web.Survey.Public** were modified to self-host a Kestrel web server. 
+
+Now, Tailspin is refactoring the Surveys application to a more granular architecture. Tailspin's motivation for refactoring is to make it easier to develop, build, and deploy the Surveys application. By decomposing the existing web and worker roles to a more granular architecture, Tailspin wants to remove the existing tightly coupled communication and data dependencies between these roles.
+
+Tailspin sees other benefits in moving the Surveys application to a more granular architecture:
+* Each service can be packaged into independent projects with a scope small enough to be managed by a small team.
+* Each service can be independently versioned and deployed.
+* Each service can be implemented using the best technology for that service.
```

Comment:
Another benefit is that each service can be independently scaled.

Reply:
Is this a microservices approach?

Reply:
Yes, but we didn't want to confuse the reader since microservices means a lot more.